app/forms/DisassociateUcr.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import play.api.data.Forms._
+import play.api.data.format.Formatter
+import play.api.data.{Form, FormError, Forms}
+import play.api.libs.json._
+import utils.validators.forms.FieldValidator._
+
+sealed abstract class DisassociateKind(val formValue: String)
+
+object DisassociateKind {
+  case object Mucr extends DisassociateKind("mucr")
+  case object Ducr extends DisassociateKind("ducr")
+
+  private val lookup = PartialFunction[String, DisassociateKind] {
+    case Mucr.formValue => Mucr
+    case Ducr.formValue => Ducr
+  }
+
+  implicit val formatter: Formatter[DisassociateKind] = new Formatter[DisassociateKind] {
```

Comment:
Still not sure what this is needed for

Reply:
As noted above - was modelled on Association.  There are now three places where we need to capture either a Mucr or Ducr in basically the same form (radio option with conditional reveal) and I think the form code could be shared.    Especially as this version does not currently cope with neither ucr value being entered.